### PR TITLE
Add govulncheck to CI

### LINKED ISSUE #971: Add govulncheck to CI
https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck

### PR DESCRIPTION
**Pull request by case-fastly**
Closes #971.

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -7,6 +7,16 @@ on:
   pull_request:
 
 jobs:
+  # Go vulnerability check
+  vuln:
+    name: govulncheck
+    runs-on: ubuntu-latest
+    steps:
+      - name: govulncheck
+        uses: golang/govulncheck-action@v1.0.1
```
Comment (cee-dub) [inline .github/workflows/go.yaml:16]: If this is set to `@v1` does it:

1. work?
2. float to the latest `v1.x.x`?

If so, we should prefer that.

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -100,8 +110,8 @@ jobs:
     timeout-minutes: 15
     strategy:
       matrix:
-        go-version: ["1.21"] # Disable Go 1.22 until TinyGo releases update
-        tinygo-version: ["0.30.0"]
+        go-version: ["1.21", "1.22"]
+        tinygo-version: ["0.31.2"]
```
Comment (cee-dub) [inline .github/workflows/go.yaml:114]: Can these be pulled out into YAML references so we don't forget to update them in both places?

**Review by cee-dub — CHANGES_REQUESTED** [review]

File: .github/workflows/go.yaml (hunk 1 above)
Comment (case-fastly) [reply .github/workflows/go.yaml:16]: Dependabot bumps these, which is why I like specifying fine-grained SemVer versions. Then we can see the diffs when they upgrade.

File: .github/workflows/go.yaml (hunk 2 above)
Comment (case-fastly) [reply .github/workflows/go.yaml:114]: Filed this in #973, "great idea, let's do it later"

**Review by cee-dub — APPROVED** [review]

**Pull request merged**